src/Payments/PaymentMethodRegistry.php

```diff
@@ -153,7 +153,8 @@ public function get_all_registered_script_data() {
 		$payment_methods = $this->get_all_active_registered();
 
 		foreach ( $payment_methods as $payment_method ) {
-			$script_data[ $payment_method->get_name() . '_data' ] = $payment_method->get_payment_method_data();
+			$script_data[ $payment_method->get_name() . '_data' ] =
+				array_merge( $payment_method->get_payment_method_data(), $payment_method->get_supported_features() );
```

Comment:
why can't this be part of `get_payment_method_data`?

Reply:
I was working on a change that required yet another filter in this place for filtering payment gateway features. You can check it out it here: https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3719/commits/a59aa05ed1cd41ace8fd67f2002e0614e0d1eaf4 Such filter is unavoidable if we want to allow PayPal integration by subscriptions. PayPal integration does not support payment gateways features required by subscriptions. Subscriptions plugin enhances the PayPal integration in a way it starts to support `subscriptions` features - that is very convoluted and it took me some time to figure it out.
I was thinking about the implementation over the weekend and I think that we should move the filter out of this place and move it just to our PayPal integration code. This will free up the logic - we will be able to remove `get_supported_features` and utilize `get_payment_method_data` for this.